resources/shared/workload-testing-utils.mjs

```diff
@@ -0,0 +1,154 @@
+
+/**
+ * BenchmarkSuitesManager
+ *
+ * A collection of test suites for a single workload.
+ */
+export class BenchmarkSuitesManager {
+    constructor(name, suites) {
+        this.name = name;
+        this.suites = suites;
+    }
+
+    getSuiteByName(name) {
+        return this.suites.find((suite) => suite.name === name);
+    }
+}
+
+/**
+ * Helper Methods
+ *
+ * Various methods that are extracted from the Page class.
+ */
+export function getParent(lookupStartNode, path) {
+    lookupStartNode = lookupStartNode.shadowRoot ?? lookupStartNode;
+    const parent = path.reduce((root, selector) => {
+        const node = root.querySelector(selector);
+        return node.shadowRoot ?? node;
+    }, lookupStartNode);
+
+    return parent;
+}
+
+export function getElement(selector, path = [], lookupStartNode = document) {
+    const element = getParent(lookupStartNode, path).querySelector(selector);
+    return element;
+}
+
+export function getAllElements(selector, path = [], lookupStartNode = document) {
+    const elements = Array.from(getParent(lookupStartNode, path).querySelectorAll(selector));
+    return elements;
+}
+
+export function forceLayout() {
+    const rect = document.body.getBoundingClientRect();
+    const e = document.elementFromPoint((rect.width / 2) | 0, (rect.height / 2) | 0);
+    return e;
+}
+
+/** **********************************************************************
+ * Benchmark Connector
+ *
+ * postMessage is used to communicate between app and benchmark.
+ * When the app os ready, an 'app-ready' message is sent to signal that the app can receive instructions.
+ *
+ * A prepare script within the apps appends window.name and window.version from the package.json file.
+ * The appId is build by appending name-version
+ * It's used as an additional safe-guard to ensure the correct app responds to a message.
+ *************************************************************************/
+export function connectToBenchmark(benchmarkSuitesManager, name, version) {
+    const appId = name && version ? `${name}-${version}` : -1;
+
+    function sendMessage(message) {
+        window.top.postMessage(message, "*");
+    }
+
+    window.onmessage = async (event) => {
+        // ensure we only let legit functions run...
+        if (event.data.id !== appId || event.data.key !== "benchmark-connector")
+            return;
+
+        switch (event.data.type) {
+            case "benchmark-suite":
+                // eslint-disable-next-line no-case-declarations
+                const params = new Params(new URLSearchParams(window.location.search));
+                // eslint-disable-next-line no-case-declarations
+                const { result } = await benchmarkSuitesManager.getSuiteByName(event.data.name).runAndRecord(params, (test) => sendMessage({ type: "step-complete", status: "success", appId, name, test }));
+                sendMessage({ type: "suite-complete", status: "success", appId, result });
+                break;
```

Comment:
nit: add a default clause with a console.error line writing the received `event.data.type`